### ISSUE TITLE
[security] Secure workflow used actions versions

### ISSUE THREAD

**Issue by gabibguti** [ampproject/amphtml#38555]
### Description

TLDR; Replace tag-pinning versions by hash-pinning versions in GitHub workflows

The workflows use other actions to perform their jobs. For example, the workflow `publish-npm-packages` uses `actions/checkout@v2` as part of the publication process. This action refers to the version by tag-pinning, as `@v2`, and not hash-pinning, as `@e2f20e631ae6d7dd3b768f56a5d2af784dd54791`. Hash-pinning is important to [mitigate the usage of compromised versions](https://github.com/ossf/scorecard/blob/main/docs/checks.md#pinned-dependencies). This request is to replace tag-pinning versions by hash-pinning versions in GitHub workflows.

### Alternatives Considered

None.

### Additional Context

Hash-pinning workflow actions versions is recommended by the OpenSSF tool Scorecards.

About me: Hi! I'm Gabriela and I work on behalf of Google and the OpenSSF suggesting supply-chain security changes to critical repositories.

**Comment by gabibguti** [comment]
Hi! Frindly ping here, if you'd like I can open a PR.
